codesign/lib/verify.dart

```diff
@@ -0,0 +1,185 @@
+  late final String _codesignId;
+  bool? _notarizationStatus;
+
+  Future<VerificationResult> run() async {
+    if (!await _codesignDisplay()) {
+      return VerificationResult.unsigned;
+    }
+    await _notarization();
+    logger.info(present());
+    return _notarizationStatus! ? VerificationResult.codesignedAndNotarized : VerificationResult.codesignedOnly;
+  }
+
+  Future<void> _notarization() async {
+    final List<String> command = <String>[
+      'codesign',
+      '--verify',
+      '-v',
+      // force online notarization check
+      '-R=notarized',
+      '--check-notarization',
+      binaryPath,
+    ];
+    final io.ProcessResult result = await pm.run(command);
+
+    // This usually means it does not satisfy notarization requirement
+    if (result.exitCode == 3 && (result.stderr as String).contains(kNotNotarizedMessage)) {
+      _notarizationStatus = false;
+      return;
+    }
+    if (result.exitCode != 0) {
+      throw Exception('''
+Command `${command.join(' ')}` failed with code ${result.exitCode}
+
+${result.stderr}
+''');
+    }
+    final String stderr = result.stderr as String;
+    if (stderr.contains('explicit requirement satisfied')) {
+      _notarizationStatus = true;
+      return;
+    }
+    throw UnimplementedError('Failed parsing the output of `${command.join(' ')}`:\n\n$stderr');
+  }
+
+  String present() {
+    return '''
+Authority:      $_codesignId
+Time stamp:     $_codesignTimestamp
+Format:         $_format
+Signature size: $_signatureSize
+Notarization:   $_notarizationStatus
+''';
+  }
+
+  /// Display overall information, intended to be machine parseable
+  ///
+  /// Output is of the format:
+  ///
+  /// Executable=/Users/developer/Downloads/mybinary
+  /// Identifier=mybinary
+  /// Format=Mach-O thin (x86_64)
+  /// CodeDirectory v=20500 size=38000 flags=0x10000(runtime) hashes=1177+7 location=embedded
+  /// Signature size=8979
+  /// Authority=Developer ID Application: Dev Shop ABC (ABCC0VV123)
+  /// Authority=Developer ID Certification Authority
+  /// Authority=Apple Root CA
+  /// Timestamp=Jan 9, 2023 at 9:39:07 AM
+  /// Info.plist=not bound
+  /// TeamIdentifier=ABCC0VV123
+  /// Runtime Version=13.1.0
+  /// Sealed Resources=none
+  /// Internal requirements count=1 size=164
+  Future<bool> _codesignDisplay() async {
+    final List<String> command = <String>[
+      'codesign',
+      '--display',
+      '-vv',
+      binaryPath,
+    ];
+    final io.ProcessResult result = await pm.run(command);
+
+    if (result.exitCode == 1) {
+      logger.severe('''
+File $binaryPath is not codesigned. To manually verify, run:
+
+codesign --display -vv $binaryPath
+''');
+      return false;
+    } else if (result.exitCode != 0) {
+      throw Exception(
+        'Command `${command.join(' ')}` failed with code ${result.exitCode}\n\n'
+        '${result.stderr}',
+      );
+    }
+
+    final List<String> lines = result.stderr.toString().trim().split('\n');
```

Comment:
Or import `dart:convert` and use `LineSplitter`.

Reply:
LineSplitter is a stream transformer while stderr is a String here (since the process already finished).

Reply:
Like this:
https://github.com/flutter/flutter/blob/2fd8250285492c7a4e216e2753f2bfd573d1b8f6/packages/flutter_tools/lib/src/android/android_device.dart#L1238-L1239

Reply:
whoah, TIL